plugins/inputs/opensearch_query/opensearch_query_test.go

```diff
@@ -0,0 +1,774 @@
+		ClientConfig: tls.ClientConfig{InsecureSkipVerify: true},
+	}
+
+	err = o.connectToOpensearch()
+	if err != nil {
+		return &container, err
+	}
+
+	// parse and build query
+	file, err := os.Open("testdata/nginx_logs")
+	if err != nil {
+		return &container, err
+	}
+	defer func(file *os.File) {
+		_ = file.Close()
+	}(file)
+
+	indexer, err := opensearchutil.NewBulkIndexer(opensearchutil.BulkIndexerConfig{
+		Client:  o.osClient,
+		Index:   testindex,
+		Refresh: "true",
+	})
+	if err != nil {
+		return &container, err
+	}
+
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		parts := strings.Split(scanner.Text(), " ")
+		size, _ := strconv.Atoi(parts[9])
+		responseTime, _ := strconv.Atoi(parts[len(parts)-1])
+
+		logline := nginxlog{
+			IPaddress:    parts[0],
+			Timestamp:    time.Now().UTC(),
+			Method:       strings.ReplaceAll(parts[5], `"`, ""),
+			URI:          parts[6],
+			Httpversion:  strings.ReplaceAll(parts[7], `"`, ""),
+			Response:     parts[8],
+			Size:         float64(size),
+			ResponseTime: float64(responseTime),
+		}
+
+		body, e := json.Marshal(logline)
+		if e != nil {
+			return &container, e
+		}
+
+		e = indexer.Add(
+			context.Background(),
+			opensearchutil.BulkIndexerItem{
+				Index:  testindex,
+				Action: "index",
+				Body:   strings.NewReader(string(body)),
+			})
+		if e != nil {
+			return &container, e
+		}
+	}
+
+	if scanner.Err() != nil {
+		return &container, err
+	}
+
+	if err = indexer.Close(context.Background()); err != nil {
+		return &container, err
+	}
+
+	return &container, nil
+}
+
+func TestOpensearchQueryIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	container, err := setupIntegrationTest(t)
+	require.NoError(t, err)
+	defer container.Terminate()
+
+	o := &OpensearchQuery{
+		URLs: []string{
+			fmt.Sprintf("https://%s:%s", container.Address, container.Ports[servicePort]),
+		},
+		Timeout:      config.Duration(time.Second * 30),
+		Log:          testutil.Logger{},
+		Username:     "admin",
+		Password:     "admin",
+		ClientConfig: tls.ClientConfig{InsecureSkipVerify: true},
+	}
+
+	err = o.connectToOpensearch()
+	require.NoError(t, err)
+
+	for _, tt := range testOpensearchAggregationData {
+		t.Run(tt.queryName, func(t *testing.T) {
+			var err error
+			var acc testutil.Accumulator
+
+			o.Aggregations = []osAggregation{tt.testAggregationQueryInput}
+			err = o.Init()
+			if (err != nil) != tt.wantInitErr {
+				t.Errorf("OpensearchQuery.Init() error = %v, wantInitErr %v", err, tt.wantInitErr)
+				return
+			} else if err != nil {
+				// Init() failures mean we're done
+				return
+			}
+
+			err = o.Gather(&acc)
+			if (len(acc.Errors) > 0) && !tt.wantQueryResErr {
+				for _, err = range acc.Errors {
+					t.Errorf("OpensearchQuery.Gather() error: %v, wantQueryResErr %v", err, tt.wantQueryResErr)
+				}
+				return
+			}
+
+			if err != nil {
+				t.Errorf("Unexpected error %v", err)
+				return
+			}
```

Comment:
```suggestion
			require.NoError(t, err)
```
Please use the require functions wherever you can.